Frends.Excel.ConvertToXML/Frends.Excel.ConvertToXML/Excel.cs

```diff
@@ -8,128 +8,104 @@
-    /// <param name="cancellationToken"></param>
-    /// <returns>Result containing the converted XML string.</returns>
-    /// <exception cref="Exception"></exception>
-    public static Result ConvertToXML(
-        [PropertyTab] Input input,
-        [PropertyTab] Options options,
-        CancellationToken cancellationToken)
+    /// <param name="cancellationToken">Token generated by Frends to stop this task.</param>
+    /// <returns>Object { bool Success, string ErrorMessage, string XML ]</returns>
+    public static Result ConvertToXML([PropertyTab] Input input, [PropertyTab] Options options, CancellationToken cancellationToken)
     {
         try
         {
             Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
 
-            using (var stream = new FileStream(input.Path, FileMode.Open))
-            {
-                using (var excelReader = ExcelReaderFactory.CreateReader(stream))
-                {
-                    var result = excelReader.AsDataSet();
-                    var xml = ConvertDataSetToXml(result, options, Path.GetFileName(input.Path), cancellationToken);
-                    return new Result(true, xml, null);
-                }
-            }
+            using var stream = new FileStream(input.Path, FileMode.Open);
+            using var excelReader = ExcelReaderFactory.CreateReader(stream);
+            var result = excelReader.AsDataSet();
+            var xml = ConvertDataSetToXml(result, options, Path.GetFileName(input.Path), cancellationToken);
+            return new Result(true, xml, null);
         }
         catch (Exception ex)
         {
             if (options.ThrowErrorOnFailure)
-            {
-                throw new Exception("Error while converting Excel file to XML", ex);
-            }
+                throw new ArgumentException("Error while converting Excel file to XML", ex);
 
             return new Result(false, null, $"Error while converting Excel file to XML: {ex}");
         }
     }
 
-    private static string ConvertDataSetToXml(DataSet result, Options options, string file_name,
-        CancellationToken cancellationToken)
+    private static string ConvertDataSetToXml(DataSet result, Options options, string file_name, CancellationToken cancellationToken)
     {
-        XmlWriterSettings settings = new XmlWriterSettings
-        {
-            OmitXmlDeclaration = true
-        };
+        XmlWriterSettings settings = new() { OmitXmlDeclaration = true };
 
         var builder = new StringBuilder();
-        using (var sw = new StringWriter(builder))
+        using var sw = new StringWriter(builder);
+        using var xw = XmlWriter.Create(sw, settings);
+
+        // Write workbook element. Workbook is also known as sheet.
+        xw.WriteStartDocument();
+        xw.WriteStartElement("workbook");
+        xw.WriteAttributeString("workbook_name", file_name);
+
+        foreach (DataTable table in result.Tables)
```

Comment:
Using LINQ here would make the iteration unnecessary complex, so I think you should suppress these warnings.